python/lsst/obs/decam/_instrument.py

```diff
@@ -71,6 +71,14 @@ def register(self, registry, update=False):
         # Combined with detector_max=100 (below), obsMax=2**25 causes the
         # number of bits in packed IDs to match the Gen2 ones.
         obsMax = 2**25
+        instrument_data_id = {
+            "name": self.getName(), "detector_max": 100, "visit_max": obsMax, "exposure_max": obsMax,
+            "class_name": get_full_type_name(self),
+            # Some schemas support default visit_system
+            "visit_system": VisitSystem.ONE_TO_ONE.value,
+        }
+        if "day_obs_offset" in registry.dimensions["instrument"].metadata:
+            instrument_data_id["day_obs_offset"] = self.day_obs_offset
```

Comment:
I don't actually see anything that sets the right value for DECam in this PR (and I see that's true of `obs_subaru` and `obs_cfht` as well).

Even though only Rubin is going to use `day_obs` as an identifier and hence care about the detailed definition, I think we need the ability to use it to make nightly coadds for precursor data too, so we want it to correspond to ~night.

To make that work, are we going to have to migrate the actual `exposure.day_obs` and `visit.day_obs` values for those instruments as well?

Reply:
HSC and CFHT are easy because in Hawaii the day changes at 2pm so no offset is fine.

I had completely forgotten where DECam is, so yes, I should probably add the 12 hour offset for it. Also yes, the observing_day for DECam that we added years ago is also going to be without an offset and so is changing in the middle of the night. Oops.